src/mem3.erl

```diff
@@ -254,6 +254,11 @@ allowed_nodes() ->
 nodes_in_zone(Nodes, Zone) ->
     [Node || Node <- Nodes, Zone == mem3:node_info(Node, <<"zone">>)].
 
+non_maintenance_nodes() ->
+    Nodes = [node() | erlang:nodes()],
+    {Modes, _} = rpc:multicall(config, get, ["couchdb", "maintenance_mode"]),
```

Comment:
@tonysun83: Let's assume you send requests to nodes in increasing order `node1, node2, node3`. But one of them delayed and you get responses in following order `resp_node1, resp_node3, resp_node2`. When you do lists:zip you would get `[{node1, resp_node1}, {node2, resp_node3}, {node3, resp_node2}]`. When you latter filter by mode you would filter out (or keep) the wrong ones.